chapter1/palindromePerm.js

```diff
@@ -0,0 +1,64 @@
+// Given a string, write a function to check if it is a permuation of a palindrome. a palindrome is a word/phrase that is the same forwards/backwrards. A permutation is a rearrangement of letters. The palindrome does not need to be limited to just dictionary words
+
+
+
+function palindromePerm(str) {
+    let strArr = str.replace(/ /g, "").toLowerCase().split(""),
+        hashStr = {},
+        leftOver = [];
+    //  assign k/v to hash
+    for (let i = 0; i < strArr.length; i++){
+        hashStr[strArr[i]] = (!hashStr.hasOwnProperty(strArr[i])) ? 1 :hashStr[strArr[i]] + 1 ;
+    }
+
+    //push keys with value of 1 to array
+    for(let k = 0; k < Object.keys(hashStr).length; k++){
```

Comment:
`Object.keys(hashStr).length` will be executed every iteration. I would assign as a variable first above the for loop